src/overlays/actors/ovl_En_Mb/z_en_mb.c

```diff
@@ -21,99 +20,1417 @@ const ActorInit En_Mb_InitVars = {
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_MORIBLIN_SPEAR_NORM);
+                }
+
+            } else if (this->actor.xzDistFromLink <= 160.0f) {
+                this->actor.speedXZ = -5.0f;
+            } else {
+                this->actor.speedXZ = 0.0f;
+            }
+        }
+        if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+            if (this->unk_32A == 0) {
+                SkelAnime_ChangeAnim(&this->skelAnime, &D_06002F10, 0.5f, 0.0f, SkelAnime_GetFrameCount(&D_06002F10), 1,
+                                     0.0f);
+                this->unk_32A = 1;
+                return;
+            }
+            yawToTarget = (Math_Vec3f_Yaw(&this->actor.posRot.pos, &this->unk_344) - this->actor.yawTowardsLink);
+
+            if (ABS(yawToTarget) < 0x4001) {
+                func_80AA68FC(this, globalCtx);
+                return;
+            }
+            func_80AA6A18(this);
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mb/func_80AA74BC.s")
+#endif
+
+void func_80AA77D0(EnMb* this, GlobalContext* globalCtx) { // Chase link
+    s32 currentFrame;
+    s16 playerInvincibilityTImer;
+
+    playerInvincibilityTImer = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+    if (playerInvincibilityTImer < 0) {
+        playerInvincibilityTImer = -playerInvincibilityTImer;
+    }
+    currentFrame = this->skelAnime.animCurrentFrame;
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06002F10);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_MORIBLIN_ATTACK);
+    }
+    if (this->unk_32E != 0) {
+        this->unk_32E--;
+        Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, this->actor.yawTowardsLink, 1, 3000, 0);
+    } else {
+        this->actor.speedXZ = 10.0f;
+        this->attackParams = 1;
+        func_80033260(globalCtx, &this->actor, &this->actor.posRot.pos, 5.0f, 3, 4.0f, 0x64, 0xF, 0);
+        if ((currentFrame != (s32)this->skelAnime.animCurrentFrame) &&
+            (((s32)this->skelAnime.animCurrentFrame == 2) || ((s32)this->skelAnime.animCurrentFrame == 6))) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_MORIBLIN_DASH);
+        }
+    }
+    if (playerInvincibilityTImer > 0x1388) {
+        this->attackParams = 0;
+        func_80AA6CC0(this);
+    }
+}
+
+Vec3f D_80AA9D50 = {
+    0.0f,
+    0.0f,
+    0.0f,
+};
+
+Vec3f D_80AA9D5C = {
+    18.0f,
+    18.0f,
+    0.0f,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mb/func_80AA77D0.s")
+typedef struct {
+    /* 0x0 */ s16 unk_0;
+    /* 0x2 */ s16 unk_2;
+    /* 0x4 */ u16 unk_4;
+} struct_80AA9D68;
 
+struct_80AA9D68 D_80AA9D68 = {
+    0x0014,
+    0x0028,
+    0x0000,
+};
+
+typedef struct {
+    s16 unk_0;
+    s16 unk_2;
+    s16 unk_4;
+} struct_80AA9D70;
+
+struct_80AA9D70 D_80AA9D70 = {
+    0xF63C,
+    0x0000,
+    0x0DAC,
```

Comment:
These aren't structs. They're `u16[3]`s declared in the body of `func_80AA7938`.